### ISSUE TITLE
Vcpkg buildfail on hdf5:x64-windows

### ISSUE THREAD

**Issue by DimitarVel** [microsoft/vcpkg#10388]
Here is the error log i get in stdout-x64-windows.log

-- Using cached C:/vcpkg/downloads/CMake-hdf5-1.10.5.tar.gz
-- Extracting source C:/vcpkg/downloads/CMake-hdf5-1.10.5.tar.gz
-- Applying patch hdf5_config.patch
-- Applying patch fix-generate.patch
-- Using source at C:/vcpkg/buildtrees/hdf5/src/hdf5-368e55e917
-- Configuring x64-windows-dbg
-- Configuring x64-windows-rel
-- Building x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:136 (message):
    Command failed: C:/CMake/bin/cmake.exe --build . --config Debug --target install -- -v
    Working Directory: C:/vcpkg/buildtrees/hdf5/x64-windows-dbg
    See logs for more information:
      C:\vcpkg\buildtrees\hdf5\install-x64-windows-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:91 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)
  ports/hdf5/portfile.cmake:50 (vcpkg_install_cmake)
  scripts/ports.cmake:90 (include)

Edit:
I am no longer getting the hdf5 error, now im getting the following error:
CMake Error at scripts/cmake/vcpkg_extract_source_archive_ex.cmake:140 (file):
  file RENAME failed to rename

    C:/vcpkg/buildtrees/boost-math/src/TEMP/math-boost-1.72.0

  to

    C:/vcpkg/buildtrees/boost-math/src/ost-1.72.0-2786b6df16

  because: No such file or directory

**Comment by linquize** [comment]
1. Can you post the content of `C:\vcpkg\buildtrees\hdf5\install-x64-windows-dbg-out.log`?
2. What version of cmake are you using? It seems you are using your own cmake, not the one downloaded by vcpkg.

**Comment by DimitarVel** [comment]
> 
> 
>     1. Can you post the content of `C:\vcpkg\buildtrees\hdf5\install-x64-windows-dbg-out.log`?
> 
>     2. What version of cmake are you using? It seems you are using your own cmake, not the one downloaded by vcpkg.

C:\vcpkg\buildtrees\hdf5\install-x64-windows-dbg-out.log

**Comment by DimitarVel** [comment]
> 
> 
>     1. Can you post the content of `C:\vcpkg\buildtrees\hdf5\install-x64-windows-dbg-out.log`?
> 
>     2. What version of cmake are you using? It seems you are using your own cmake, not the one downloaded by vcpkg.

[1/321] C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1424~1.283\bin\Hostx64\x64\cl.exe   -D_BIND_TO_CURRENT_VCLIBS_VERSION=1 -D_CONSOLE -D_CRT_SECURE_NO_WARNINGS -IC:\vcpkg\installed\x64-windows\include -IC:\vcpkg\buildtrees\hdf5\src\hdf5-368e55e917\hdf5-1.10.5\src -I. /DWIN32 /D_WINDOWS /W3 /utf-8 /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /Fosrc\CMakeFiles\H5make_libsettings.dir\H5make_libsettings.c.obj /Fdsrc\CMakeFiles\H5make_libsettings.dir\ /FS -c C:\vcpkg\buildtrees\hdf5\src\hdf5-368e55e917\hdf5-1.10.5\src\H5make_libsettings.c
Microsoft (R) C/C++ Optimizing Compiler Version 19.24.28319 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

[2/321] C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1424~1.283\bin\Hostx64\x64\cl.exe   -D_BIND_TO_CURRENT_VCLIBS_VERSION=1 -D_CONSOLE -D_CRT_SECURE_NO_WARNINGS -IC:\vcpkg\installed\x64-windows\include -IC:\vcpkg\buildtrees\hdf5\src\hdf5-368e55e917\hdf5-1.10.5\src -I. /DWIN32 /D_WINDOWS /W3 /utf-8 /MP  /D_DEBUG /MDd /Z7 /Ob0 /Od /RTC1 /showIncludes /Fosrc\CMakeFiles\H5detect.dir\H5detect.c.obj /Fdsrc\CMakeFiles\H5detect.dir\ /FS -c C:\vcpkg\buildtrees\hdf5\src\hdf5-368e55e917\hdf5-1.10.5\src\H5detect.c
Microsoft (R) C/C++ Optimizing Compiler Version 19.24.28319 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

[3/321] cmd.exe /C "cd . && C:\CMake\bin\cmake.exe -E vs_link_exe --intdir=src\CMakeFiles\H5make_libsettings.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100183~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100183~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1424~1.283\bin\Hostx64\x64\link.exe  src\CMakeFiles\H5make_libsettings.dir\H5make_libsettings.c.obj  /out:bin\H5make_libsettings.exe /implib:bin\H5make_libsettings.lib /pdb:bin\H5make_libsettings.pdb /version:0.0  /machine:x64 /debug /INCREMENTAL /subsystem:console  ws2_32.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cd ."
Microsoft (R) Incremental Linker Version 14.24.28319.0
Copyright (C) Microsoft Corporation.  All rights reserved.

Microsoft (R) Incremental Linker Version 14.24.28319.0
Copyright (C) Microsoft Corporation.  All rights reserved.

[4/321] cmd.exe /C "cd . && C:\CMake\bin\cmake.exe -E vs_link_exe --intdir=src\CMakeFiles\H5detect.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100183~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100183~1.0\x64\mt.exe --manifests  -- C:\PROGRA~2\MICROS~1\2019\COMMUN~1\VC\Tools\MSVC\1424~1.283\bin\Hostx64\x64\link.exe  src\CMakeFiles\H5detect.dir\H5detect.c.obj  /out:bin\H5detect.exe /implib:bin\H5detect.lib /pdb:bin\H5detect.pdb /version:0.0  /machine:x64 /debug /INCREMENTAL /subsystem:console  ws2_32.lib  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib && cd ."
Microsoft (R) Incremental Linker Version 14.24.28319.0
Copyright (C) Microsoft Corporation.  All rights reserved.

Microsoft (R) Incremental Linker Version 14.24.28319.0
Copyright (C) Microsoft Corporation.  All rights reserved.

[5/321] cmd.exe /C "cd /D C:\vcpkg\buildtrees\hdf5\x64-windows-dbg && C:\vcpkg\buildtrees\hdf5\x64-windows-dbg\bin\H5detect.exe C:/vcpkg/buildtrees/hdf5/x64-windows-dbg/H5Tinit.c && C:\CMake\bin\cmake.exe -E touch C:/vcpkg/buildtrees/hdf5/x64-windows-dbg/gen_SRCS.stamp1"
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             [6/321] cmd.exe /C "cd /D C:\vcpkg\buildtrees\hdf5\x64-windows-dbg && C:\CMake\bin\cmake.exe -E copy_if_different C:/vcpkg/buildtrees/hdf5/x64-windows-dbg/H5Tinit.c C:/vcpkg/buildtrees/hdf5/x64-windows-dbg/shared/H5Tinit.c && C:\CMake\bin\cmake.exe -E touch C:/vcpkg/buildtrees/hdf5/x64-windows-dbg/shared/shared_gen_SRCS.stamp1"

**Comment by LilyWangL** [comment]
@DimitarVel Could you remove the directory _C:/vcpkg/buildtrees/boost-math/src_ and install hdf5 again? And the port hdf5 doesn't depends on boost-math.

**Comment by DimitarVel** [comment]
@LilyWangL I get errors for both boost-math and hdf5. I tried to delete the src folders of both and retry but that did not fix it. I also tried git pull && vcpkg update but that didnt work either. 

The error message is as follows: 

`
file RENAME failed to rename

    C:/vcpkg/buildtrees/boost-math/src/TEMP/math-boost-1.72.0

  to

    C:/vcpkg/buildtrees/boost-math/src/ost-1.72.0-2786b6df16

  because: No such file or directory

Call Stack (most recent call first):
  scripts/cmake/vcpkg_from_github.cmake:139 (vcpkg_extract_source_archive_ex)
  ports/boost-math/portfile.cmake:5 (vcpkg_from_github)
  scripts/ports.cmake:90 (include)
`

If you look at the directories closely, the 2 files are in 2 different directories which makes sense why the error would occur but I dont see why it is installing files in a TEMP folder and not in the src directly.
